File: src/Branch/Decorator/Decorator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FluentBehaviorTree
{
    public abstract class Decorator : Branch
    {
        public Node child;

        public override void AddChild(Node n)
        {
            child = n;
        }
    }
}

File: C++/BehaviorTreesLib/Decorator.cpp
#include "Decorator.h"

namespace fluentBehaviorTree
{
	void Decorator::addChild(Node & n)
	{
		mChild = &n;
	}

	Decorator::~Decorator()
	{
		if (mChild != nullptr)
		{
			delete mChild;
			mChild = nullptr;
		}
	}
}